account_id.go

```diff
@@ -59,7 +59,7 @@ func AccountIDFromSolidityAddress(s string) (AccountID, error) {
 }
 
 func (id *AccountID) Validate(client *Client) error {
-	if !id.isZero() && client != nil && client.networkName != nil {
+	if !id.isZero() && client != nil && client.networkName != nil && client.autoValidateChecksums {
```

Comment:
`client.autoValidateChecksums` should not be part of the `*ID.Validate()` methods. The point of `*ID.Validate()` methods is to allow the user to force validation of a certain ID. `Client.autoValidateChecksums` on the other hand is strictly meant set if `*.Execute()` should validate checksums of all IDs or not.

Reply:
oooo